internal/service/project/model_project.go

```diff
@@ -202,3 +206,27 @@ func UpdateProjectBool(plan, state types.Bool, setting **bool) bool {
 	}
 	return false
 }
+
+func NewTFTags(tags []admin.ResourceTag) types.Map {
+	typesTags := make(map[string]attr.Value, len(tags))
+	for _, tag := range tags {
+		typesTags[tag.Key] = types.StringValue(tag.Value)
+	}
+	return types.MapValueMust(types.StringType, typesTags)
+}
+
+func NewResourceTags(ctx context.Context, tags types.Map) []admin.ResourceTag {
+	if tags.IsNull() || len(tags.Elements()) == 0 {
+		return []admin.ResourceTag{}
+	}
+	elements := make(map[string]types.String, len(tags.Elements()))
+	_ = tags.ElementsAs(ctx, &elements, false)
```

Comment:
This is done 8 times in the resource_project.go and it is never checked 😅
E.g., https://github.com/mongodb/terraform-provider-mongodbatlas/blob/CLOUDP-242261_project_tags/internal/service/project/resource_project.go#L353
Not sure if we should add the check here. WDYT?

Reply:
If it fails and we ignore the error, in this case `elements` will not be populated and tagsAdmin will remain empty. Not sure about the other 8 occurances. I think it would be better to do something like [this](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/78c807dbef5db7750ff384d36ee84cfb002dbe4c/internal/service/streamconnection/model_stream_connection.go#L46).
Can be addressed in another PR since this is already a big one.